airbyte-webapp/src/views/Connector/ConnectorDocumentationLayout/DocumentationPanel.tsx

```diff
@@ -58,20 +66,60 @@ export const DocumentationPanel: React.FC = () => {
     setDocumentationPanelOpen(false);
   }, [setDocumentationPanelOpen, location.pathname]);
 
+  const [activeTab, setActiveTab] = useState("docs");
+  const tabs = [
+    {
+      id: "docs",
+      name: <FormattedMessage id="sources.documentationPanel.tabs.docs" />,
+    },
+    {
+      id: "schema",
+      name: <FormattedMessage id="sources.documentationPanel.tabs.schema" />,
+    },
+    {
+      id: "erd",
+      name: <FormattedMessage id="sources.documentationPanel.tabs.erd" />,
+    },
+  ];
+
   return isLoading || documentationUrl === "" ? (
     <LoadingPage />
   ) : (
     <div className={styles.container}>
       <PageHeader withLine title={<FormattedMessage id="connector.setupGuide" />} />
-      <Markdown
-        className={styles.content}
-        content={
-          docs && !error
-            ? prepareMarkdown(docs, isCloudApp() ? "cloud" : "oss")
-            : formatMessage({ id: "connector.setupGuide.notFound" })
-        }
-        rehypePlugins={urlReplacerPlugin}
-      />
+      {isSource && (
+        <StepsMenu
+          className={styles.stepsContainer}
+          lightMode
+          data={tabs}
+          onSelect={setActiveTab}
+          activeStep={activeTab}
+        />
+      )}
+
+      {match(activeTab)
```

Comment:
I've added `ts-pattern` because I think it's a great library and allow us to have exahustive pattern matching and also cleaner/more readable code

Reply:
You are right, in this case it's not a huge benefit  but still I think it makes code so much easy to read. You can see in a nutshell the different branches that depends on a given variable value and also if you are forgetting any possibility. While  the standard way I feel works for smaller use cases or cases where we just create new components and we can keep cases small in 2-3 lines.

Where I see this being helpful and helping as establishing a clean pattern is managing request states. Eg: https://github.com/airbytehq/airbyte/pull/21861/files#diff-c298f10fce8256b2dc3a4f848e47ed70b7b574f0ba0bc8793e8fccb238d1d8eeR25 

I can def present it in the next chapter sync to gather opinions from the team.
This is the first time I've "added" a new pattern so wasn't really sure about the approach that I should follow.